src/appleseed-max-impl/appleseedinteractive/interactiverenderercontroller.h

```diff
@@ -80,6 +86,101 @@ class CameraObjectUpdateAction
     renderer::Project&                                m_project;
 };
 
+class MaterialUpdateAction
+  : public ScheduledAction
+{
+  public:
+    MaterialUpdateAction(
+        renderer::Project&      project,
+        const IAppleseedMtlMap& material_map)
+      : m_material_map(material_map)
+      , m_project(project)
+    {
+    }
+
+    void update() override
```

Comment:
I would suggest to implement this class (and possibly the other classes in this file) in the .cpp file instead. That should help decoupling and allow to remove some `#include` from this file, hopefully.